packages/admin-ui/ui/src/domain/product-categories/modals/edit-product-category.tsx

```diff
@@ -1,80 +1,142 @@
-import { NextSelect } from "../../../components/molecules/select/next-select"
-import useNotification from "../../../hooks/use-notification"
-import { Option } from "../../../types/shared"
-import { getErrorMessage } from "../../../utils/error-messages"
-import TreeCrumbs from "../components/tree-crumbs"
+import { useEffect } from "react";
+
+import { ProductCategory } from "@medusajs/medusa";
+import { useAdminUpdateProductCategory } from "medusa-react";
+import { TFunction } from "i18next";
+import { useTranslation } from "react-i18next";
+
+import Button from "../../../components/fundamentals/button";
+import CrossIcon from "../../../components/fundamentals/icons/cross-icon";
+import InputField from "../../../components/molecules/input";
+import TextArea from "../../../components/molecules/textarea";
+import SideModal from "../../../components/molecules/modal/side-modal";
+import { NextSelect } from "../../../components/molecules/select/next-select";
+import useNotification from "../../../hooks/use-notification";
+import { Option } from "../../../types/shared";
+import { getErrorMessage } from "../../../utils/error-messages";
+import TreeCrumbs from "../components/tree-crumbs";
+import MetadataForm, {
+  getSubmittableMetadata,
+} from "../../../components/forms/general/metadata-form";
+import { Controller, useForm } from "react-hook-form";
+import { nestedForm } from "../../../utils/nested-form";
+import { CategoryFormData } from "./add-product-category";
 
 const visibilityOptions: (t: TFunction) => Option[] = (t) => [
   {
     label: "Public",
     value: "public",
   },
   { label: "Private", value: "private" },
-]
+];
 
 const statusOptions: (t: TFunction) => Option[] = (t) => [
   { label: "Active", value: "active" },
   { label: "Inactive", value: "inactive" },
-]
+];
 
 type EditProductCategoriesSideModalProps = {
-  activeCategory: ProductCategory
-  close: () => void
-  isVisible: boolean
-}
+  activeCategory: ProductCategory;
+  close: () => void;
+  isVisible: boolean;
+  categories: ProductCategory[];
+};
 
 /**
  * Modal for editing product categories
  */
 function EditProductCategoriesSideModal(
   props: EditProductCategoriesSideModalProps
 ) {
-  const { isVisible, close, activeCategory, categories } = props
+  const { isVisible, close, activeCategory, categories } = props;
 
-  const [name, setName] = useState("")
-  const [handle, setHandle] = useState("")
-  const [description, setDescription] = useState("")
-  const [isActive, setIsActive] = useState(true)
-  const [isPublic, setIsPublic] = useState(true)
-
-  const { t } = useTranslation()
-  const notification = useNotification()
+  const { t } = useTranslation();
+  const notification = useNotification();
 
   const { mutateAsync: updateCategory } = useAdminUpdateProductCategory(
     activeCategory?.id
-  )
+  );
+
+  const form = useForm<CategoryFormData>({
+    defaultValues: {
+      name: activeCategory?.name || "",
```

Comment:
todo: remove `?.` and `|| ""`

the type should be true, this indicates it's not the case, we should then only render the modal if activeCategory is set (or return null early from here)